Classes/Emogrifier.php

```diff
@@ -681,8 +692,22 @@ private function getUnifiedHtml()
         } else {
             $bodyWithoutUnprocessableTags = $this->html;
         }
+        
+        if ($this->shouldFixMissingCharsetDeclaration) {
+            if (!stristr($bodyWithoutUnprocessableTags, 'Content-Type')) {
+
+                // We are trying to insert the meta tag to the right spot in the DOM as if we just prepend it to the HTML we will lose attributes set to the HTML tag.
+                if (stristr($bodyWithoutUnprocessableTags, '<head')) {
```

Comment:
For stristr checks, please always use !== false as the return value isn't always a boolean.